### ISSUE TITLE
Waydroid Android 11 Zygisk not enabled

### ISSUE THREAD

**Issue by pagkly** [topjohnwu/Magisk#5501]
Device: Waydroid x86 LineageOS 18
Android version: 11
Magisk version name: Canary
Magisk version code: 1a1b346c (24102)

[magisk_log_2022-03-01T09_39_35.log](https://github.com/topjohnwu/Magisk/files/8160562/magisk_log_2022-03-01T09_39_35.log)

Magisk Canary is installed. Root manager is working.
However, zygisk still says no after enabling and reboot. 
Magisk Modules successfully installed (it is inside the modules folder) but not loading and missing from the modules list.
![Screenshot from 2022-03-01 20-47-16](https://user-images.githubusercontent.com/28628331/156145817-d84074f6-79d7-4bcb-a17a-e06a53fec002.png)
![Screenshot from 2022-03-01 20-47-38](https://user-images.githubusercontent.com/28628331/156145851-25c59ab8-6165-4786-ba50-bbfdff506c3a.png)

**Comment by yujincheng08** [comment]
full bootlog required

**Comment by pagkly** [comment]
@yujincheng08 Hi thank you for your reply. I have since tried with the Canary 24103 build but it didnt get installed properly somehow. How do I download Canary 24102 [https://github.com/topjohnwu/Magisk/commit/1a1b346c05ff6ab8e4c3b55155d3f2f69205003c]() since it was the last one I got installed and working.

**Comment by pagkly** [comment]
> I/Magisk ( 49): Magisk [1a1b346](https://github.com/topjohnwu/Magisk/commit/1a1b346c05ff6ab8e4c3b55155d3f2f69205003c)(24101) daemon started
> 
> Definitely useless log.

Does this mean zygisk will not work?

**Comment by pagkly** [comment]
Let me try again with 24201. Its just released today so I have not tried them.
I need to unpack system.img again and transfer magisk manually so it takes some time to apply everything.

**Comment by pagkly** [comment]
@yujincheng08 
I finally got 24201 installed.
Here is the logs:
[magisk.log](https://github.com/topjohnwu/Magisk/files/8170660/magisk.log)
[liveboot.log](https://github.com/topjohnwu/Magisk/files/8170661/liveboot.log)
Let me know what you think.

**Comment by pagkly** [comment]
```
:/ # su -c cat /proc/1/mountinfo | grep /system/bin/app_process                                                                                                                                     
805 799 7:0 /system/bin/app_process32 /system/bin/app_process32 ro,relatime shared:446 - ext4 /dev/magisk/.magisk/block/system_root ro,seclabel
806 799 7:0 /system/bin/app_process64 /system/bin/app_process64 ro,relatime shared:446 - ext4 /dev/magisk/.magisk/block/system_root ro,seclabel
1021 805 0:63 /.magisk/zygisk/app_process32 /system/bin/app_process32 rw,relatime shared:444 - tmpfs tmpfs rw,seclabel,mode=755,inode64
1022 797 0:63 /.magisk/zygisk/app_process32 /dev/magisk/.magisk/mirror/system_root/system/bin/app_process32 rw,relatime shared:444 - tmpfs tmpfs rw,seclabel,mode=755,inode64
1023 806 0:63 /.magisk/zygisk/app_process64 /system/bin/app_process64 rw,relatime shared:444 - tmpfs tmpfs rw,seclabel,mode=755,inode64
1024 797 0:63 /.magisk/zygisk/app_process64 /dev/magisk/.magisk/mirror/system_root/system/bin/app_process64 rw,relatime shared:444 - tmpfs tmpfs rw,seclabel,mode=755,inode64
```

**Comment by pagkly** [comment]
@yujincheng08 
Just to update, this is the latest log and init.rc.
Let me know what you think.
[magisk.log](https://github.com/topjohnwu/Magisk/files/8175044/magisk.log)
[liveboot.log](https://github.com/topjohnwu/Magisk/files/8175045/liveboot.log)
[initrc.txt](https://github.com/topjohnwu/Magisk/files/8175047/initrc.txt)

**Comment by HuskyDG** [comment]
He will never care about Magisk on Emulator so any bugreports about Magisk on emulator will be ignore

**Comment by HuskyDG** [comment]
Because this is not a bugs from actual devices so developer will not care about this at all

**Comment by yujincheng08** [comment]
The situation is, zygote seems not actually to use the app_process mounted by magisk. Please check why by yourself. You may check the inode of `/proc/$(pidof zygote64)/exe` and `/data/adb/magisk64`. They should be the same.

**Comment by HuskyDG** [comment]
> The situation is, zygote seems not actually to use the app_process mounted by magisk. Please check why by yourself. You may check the inode of `/proc/$(pidof zygote64)/exe` and `/data/adb/magisk64`. They should be the same.

You mean `/data/adb/magisk/magisk64` or `/data/adb/magisk`?

**Comment by pagkly** [comment]
@yujincheng08 Sorry checking inode is with `stat` or `ls -li` right?
![Screenshot from 2022-03-03 23-38-56](https://user-images.githubusercontent.com/28628331/156566337-533f0a3a-8023-473b-abc6-d6f981a620cc.png)

**Comment by HuskyDG** [comment]
`/proc/$(pidof zygote64)/exe` symlink to `/system/bin/app_process`

**Comment by yujincheng08** [comment]
Well, my bad. Whenever zygisk is launched, exe will change to the real app_process. But the key point is still. Magisk's app_process is not launched by init for starting zygote.

**Comment by HuskyDG** [comment]
> 

from dmesg it shows: 
```
Could not find '/dev/magisk/.magisk/zygisk/app_process64' for ctl.start
```

**Comment by pagkly** [comment]
@HuskyDG based on what the dmesg showed I was thinking it meant that app_process64 wont launch with `start` command and probably with `exec` or `trigger`

**Comment by pagkly** [comment]
@yujincheng08 oh ill probably comment on the other report regarding my Samsung Galaxy Note 3 (hlte) but I dont know which issue it would be under. On my hlte device when I tried :

> canary 24201 -> cannot mount  error like in #5530

> v24.2 -> It boots slow more than 1 minute but when it does zygisk not enabled. No lines containing  zygisk/app_process from magisk.log so zygisk not starting. Hlte devices has similar chipset and kernel to Nexus 5 and its quite similar to issue #5516 but in my case it booted up just very slow.

> v24.1 was the last one that has zygisk working in my Note 3.

**Comment by pagkly** [comment]
@yujincheng08
I managed to get zygote strace log.
I hope you can check them out and let me know what you think is the problem.
[zygote.txt](https://github.com/topjohnwu/Magisk/files/8190802/zygote.txt)

**Comment by Erisfiregamer1** [comment]
Hey, I'm aware this isn't what the issue is about, but can you tell me how you got Magisk installed to the image? I can't find how.

**Comment by pagkly** [comment]
> Hey, I'm aware this isn't what the issue is about, but can you tell me how you got Magisk installed to the image? I can't find how.

you can check my repo [magisk on waydroid](https://github.com/pagkly/MagiskOnWaydroid)

**Comment by baptx** [comment]
I got Zygisk enabled here:
https://github.com/casualsnek/waydroid_script/pull/143#issuecomment-1893679352

**Comment by DavidFirefox** [comment]
@baptx thanks for your information.
Could you be more precise ?
You have test with which version on waydroid ?
Where do you find  "Direct Install into system partition" 
I'm not sure to correctly understand.
Thanks in advance

**Comment by baptx** [comment]
@DavidFirefox I used Waydroid with Android 11.
It was a long time ago so I don't remember the details but the README from https://github.com/casualsnek/waydroid_script also mentions "Direct Install into system partition".
You can create a new issue so others can help if needed.

**Comment by mymirai-nikki** [comment]
Hi @baptx 

> Anyway I don't need Magisk since I was not able to bypass SafetyNet / Play Integrity on Waydroid.

does that mean you somehow managed to bypass SafetyNet / Play Integrity on other Android emulator? I'd like to know! ^^

**Comment by baptx** [comment]
Hi @mymirai-nikki, no, I think the issue is the same on Android-x86 and Anbox (which is deprecated now). And I did not try Bliss OS yet.
Here is the dedicated issue to see if people get it working: https://github.com/waydroid/waydroid/issues/182
